src/screens/Forum/ForumHomeScreen.js

```diff
@@ -26,56 +35,138 @@ export default class ForumHomeScreen extends React.Component {
   }
 
+  handleEndReached() {
+    const { allPostsLoaded } = this.state;
+
+    if (!allPostsLoaded) {
+      this.loadMore();
+    }
+  }
+
+  // Fetch more data from firestore to load next posts
+  loadMore() {
+    this.setState({ loadingMore: true });
+    const { postLimit, forumPosts, lastReferencedPost } = this.state;
+
+    firestore().collection('forum_posts')
+      .orderBy('createdAt', 'desc')
+      .startAfter(lastReferencedPost.createdAt)
+      .limit(postLimit)
+      .get()
+      .then((snapshot) => {
+        const newForumPosts = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
+        const newLastReferenced = newForumPosts[newForumPosts.length - 1];
+
+        if (newForumPosts.length === 0) {
+          this.setState({ allPostsLoaded: true, loadingMore: false });
+        } else {
+          this.setState({ // Store/append updated data for next postLimit# of posts in state
+            forumPosts: [...forumPosts, ...newForumPosts],
+            lastReferencedPost: newLastReferenced,
+          });
+        }
+      })
+      .catch((error) => this.setState({ errorMessage: error.message, loadingMore: false }));
+  }
+
+  renderHeaderComponent() {
+    const { loading, errorMessage } = this.state;
+
+    if (loading && !errorMessage) {
+      return (
+        <View style={styles.activityIndicator}>
+          <ActivityIndicator />
+        </View>
+      );
+    }
+    if (errorMessage && !loading) {
+      return (
+        <Text style={{ color: 'red' }}>
+          {errorMessage}
+        </Text>
+      );
+    }
+    if (errorMessage && loading) {
+      return (
+        <View>
+          <View style={styles.activityIndicator}>
+            <ActivityIndicator />
+          </View>
+          <Text style={{ color: 'red' }}>
+            {errorMessage}
+          </Text>
+        </View>
+      );
+    }
+    return null;
+  }
+
   render() {
     const {
-      forumPosts, loading, errorMessage, currentUserID,
+      forumPosts, currentUserID, loadingMore,
     } = this.state;
     const { navigation } = this.props;
 
     return (
-      <View style={styles.homeContainer}>
-        <ScrollView>
-          {loading && <ActivityIndicator /> }
-          {errorMessage && <Text style={{ color: 'red' }}>{errorMessage}</Text>}
-          {forumPosts.map((post) => {
-            const date = post.createdAt ? post.createdAt.toDate() : null;
-            const time = date ? date.toTimeString() : null;
-
-            return (
-              <ForumPost
-                belongsToCurrentUser={currentUserID === post.userID}
-                key={post.id}
-                userID={post.userID ? post.userID : null}
-                time={time}
-                postID={post.id}
-                navigateToPostScreen={this.navigateToPostScreen}
-              >
-                {post.title}
-              </ForumPost>
-            );
+      <View style={styles.mainContainer}>
+        <FlatList
```

Comment:
What was the reasoning behind the transition from ScrollView to FlatList?

Reply:
(from previous pr) I don't remember exactly, but FlatList served the same purpose and was better suited to rendering, loading the components, and detecting events than Scrollview.

Reply:
Sounds good!